lib/hbase/hbase-client.js

```diff
@@ -244,14 +244,33 @@ HbaseClient.prototype.getAccountPayments = function (options, callback) {
   var startRow = options.account + '|' + utils.formatTime(options.start);
   var endRow   = options.account + '|' + utils.formatTime(options.end);
 
-  this.getScan({
-    table      : table,
-    startRow   : startRow,
-    stopRow    : endRow,
-    limit      : options.limit,
-    descending : options.descending
-  }, function (err, rows) {
-    callback(err, formatPayments(rows || []));
+  if(options.currency) {
+    options.currency= options.currency.toUpperCase();
+  }
+
+  if(options.type) {
+    options.type= options.type.toLowerCase();
+  }
+
+  var maybeFilters =
+  [{ qualifier: 'currency', value: options.currency, family: 'f', comparator: '=' },
+   { qualifier: options.type, value: options.account, family: 'f', comparator: '=' }];
+
+  var filterString= this.buildSingleColumnValueFilters(maybeFilters);
+
+  this.getScanWithMarker(this, {
+    table        : table,
+    startRow     : startRow,
+    stopRow      : endRow,
+    limit        : options.limit,
+    descending   : options.descending,
+    marker       : options.marker,
+    type         : options.type,
```

Comment:
Rugit, type is used when building the filter string and isn't used in getScan directly. Should be removed indeed.